whois-api/src/main/java/net/ripe/db/whois/api/rdap/RedactionObjectMapper.java

```diff
@@ -0,0 +1,65 @@
+package net.ripe.db.whois.api.rdap;
+
+import com.google.common.collect.Lists;
+import net.ripe.db.whois.api.rdap.domain.Redaction;
+import net.ripe.db.whois.api.rdap.domain.Role;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.springframework.stereotype.Component;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import static net.ripe.db.whois.common.rpsl.AttributeType.NOTIFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MBRS_BY_REF;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_DOMAINS;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_LOWER;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_REF;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_ROUTES;
+
+@Component
+public class RedactionObjectMapper {
+
+    private static final List<AttributeType> unsupportedRegistrantAttributes = Lists.newArrayList(MBRS_BY_REF,
+            MNT_DOMAINS, MNT_LOWER, MNT_REF, MNT_ROUTES);
+
+    private static final List<AttributeType> unsupportedPersonalAttributes = Lists.newArrayList(NOTIFY);
+
+    public static String REDACTED_ENTITIES_SYNTAX = "$.entities[?(@.handle=='%s')]";
+
+    public static String UNSUPPORTED_VCARD_SYNTAX = "$.entities[?(@.roles=='%s')].vcardArray[1][?(@[0]=='%s')]";
+
+    public static Set<Redaction> createEntityRedaction(final List<RpslAttribute> rpslAttributes){
+        return rpslAttributes.stream()
+                .filter( rpslAttribute -> unsupportedRegistrantAttributes.contains(rpslAttribute.getType()))
+                .flatMap( rpslAttribute -> {
+                    final List<Redaction> redactions = Lists.newArrayList();
+                    for (final CIString values : rpslAttribute.getCleanValues()) {
+                        redactions.add(createRedactionByAttributeType(AttributeType.getByName(rpslAttribute.getKey()),
```

Comment:
AttributeType.getByName(rpslAttribute.getKey()) is rpslAttribute.getType gives you same result?